src/Listeners/PauseBeam.php

```diff
@@ -15,15 +17,26 @@ class PauseBeam implements ShouldQueue
      */
     public function handle(PlatformBroadcastEvent $event): void
     {
-        Beam::where('collection_chain_id', $event->broadcastData['collectionId'])
-            ->get()
-            ->each(function ($beam) {
-                $beam->update(['flags_mask' => BeamService::getFlagsValue(
-                    collect(array_merge($beam->flags ?? [], ['PAUSED']))
-                        ->unique()
-                        ->map(fn ($flag) => ['flag' => $flag, 'enabled' => true])->all()
-                )]);
-                Log::info("Pausing beam {$beam->code} cause the collection {$beam->collection_chain_id} was paused.");
-            });
+        foreach (Beam::where('collection_chain_id', $event->broadcastData['collectionId'])->get() as $beam) {
+            $hasTransfers = BeamClaim::claimable()
+                ->where('type', BeamType::TRANSFER_TOKEN->name)
+                ->where('beam_id', $beam->id)
+                ->exists();
+            $hasMints = BeamClaim::claimable()
+                ->where('type', BeamType::MINT_ON_DEMAND->name)
+                ->where('beam_id', $beam->id)
+                ->exists();
+            if (($hasMints && !$hasTransfers) || ($hasMints && $hasTransfers)) {
```

Comment:
```suggestion
            if ($hasMints) {
```
Isn't this logically equivalent since you're checking for $hasMints in both?  If so then I think you can remove the $hasTransfers query and simplify the if statement to just checking `$hasMints`